include/swift/AST/DiagnosticsSema.def

```diff
@@ -1054,6 +1054,10 @@ NOTE(single_confusable_character,none,
       (bool, StringRef, StringRef, StringRef, StringRef))
 ERROR(cannot_find_type_in_scope,none,
       "cannot find type %0 in scope", (DeclNameRef))
+ERROR(cannot_find_type_in_cast_expression,none,
+      "type-casting operator expects a type on its right-hand side (got: %0)", (const ValueDecl *))
```

Comment:
Could you try `%kind0 %0` here instead of just `%0`?

Reply:
you mean `%kind0` right? Done. Now it presents "parameter a" rather than "a".